src/pages/index.js

```diff
@@ -96,142 +103,152 @@ popUpWithImage.setEventListeners();
 );
 popUpEditProfile.setEventListeners();
 
 //PopUpWithForm
 const popUpAddItem = new PopUpWithForm("#modal-add-profile", (formData) => {
-  return new Promise ((resolve, reject) => {
+  return new Promise((resolve, reject) => {
     const name = formData.name;
     const link = formData.link;
     handleAddProfileFormSubmit(name, link)
-    .then(() => {
-      popUpAddItem.close();
-      resolve();
-  })
-  .catch((err) => {
-    console.error(err);
-    reject(err);
-   })
-  })
+      .then(() => {
+       // popUpAddItem.close();
+        resolve();
+      })
+      .catch((err) => {
+        console.error(err);
+        reject(err);
+      });
+  });
 });
 popUpAddItem.setEventListeners();
 
 //PopUpWithForm
 const popUpAvatar = new PopUpWithForm("#modal-avatar", (formData) => {
   //const avatarLink = cardData.avatar;
   const avatar = formData.avatar;
-  return new Promise ((resolve, reject) => {
-    api.updateAvatar(avatar)
-    .then((updatedAvatar) => {
-      userinfo.setAvatar(updatedAvatar.avatar);
-      popUpAvatar.close();
-      resolve();
-    })
-    .catch((err) => {
-      console.error(err);
-      reject(err);
-    })
-  })
+  return new Promise((resolve, reject) => {
+    api
+      .updateAvatar(avatar)
+      .then((updatedAvatar) => {
+        userinfo.setAvatar(updatedAvatar.avatar);
+        resolve();
+        popUpAvatar.close();
+      })
+      .catch((err) => {
+        console.error(err);
+        reject(err);
+      });
+  });
 });
 popUpAvatar.setEventListeners();
 
 //PopUpWithConfirmation
-const popUpConfirm = new PopUpWithConfirmation("#modal-confirm-delete")
+const popUpConfirm = new PopUpWithConfirmation("#modal-confirm-delete");
 popUpConfirm.setEventListeners();
 
-
 //Event Handlers
 function handleAddProfileFormSubmit(title, link) {
   return new Promise((resolve, reject) => {
-  api.createCard({ name: title, link })
-  .then((card) => {
-    section.addItem(createCard(card));
-    console.log("CARD CHECK! please please work T_T.", card);
-    resolve(); 
-  })
-.catch((err) => {
-  console.error(err);
-  reject(err);
-}) 
-.finally(() => {
-  popUpAddItem.close();
-})
+    api
+      .createCard({ name: title, link })
+      .then((card) => {
+        section.addItem(createCard(card));
+        //console.log("CARD CHECK! please please work T_T.", card);
+        resolve();
+        popUpAddItem.close();
+      })
+      .catch((err) => {
+        console.error(err);
+        reject(err);
+      })
   });
-
 }
 
-
 function handleImageClick(data) {
   popUpWithImage.open(data);
 }
 
 function handleTrashButtonClick(item) {
-  console.log("enter trash index.js")
-  popUpConfirm.setLoading(true);
-  popUpConfirm.setSubmitCall(() => { 
-    api.deleteCard(item.getId())
-    .then(() => {
-      //const cardElement = document.querySelector(`[data-card-id="${cardId}"]`);
-      //if (cardElement){
-        popUpConfirm.close();
+  console.log("enter trash index.js");
+  popUpConfirm.setDeleting(true);
```

Comment:
`popUpConfirm.setDeleting(true);` should be in line 193 **inside** the request function to correctly show loading